core/sql/generator/GenResources.cpp

```diff
@@ -97,92 +77,74 @@ static ExScratchDiskDrive * genScratchDriveList(const NAString &def,
   LIST(ExScratchDiskDrive *) tempList;
   CollHeap *heap = generator->wHeap();
   Space *space = generator->getSpace();
+  struct stat st;
+  
+ 
+  Lng32 nodeNum;
+  char *token = new(heap) char[PATH_MAX];
 
-  // ---------------------------------------------------------------------
-  // process the NT default
-  // ---------------------------------------------------------------------
-  while (str && *str)
+  char *sep = (char *)":";
+  token = strtok((char *)str,sep);
+  while (token != NULL)
     {
-      Lng32 nodeNum;
-      char *driveLetter = new(heap) char[2];
-
-      driveLetter[1] = 0;
-      if (ValidateDiskListNT::getNextDriveLetterAndAdvance(
-	   str,nodeNum,driveLetter[0]))
-	{
-	  // syntax error in default, issue a warning (not an error)
-	  *CmpCommon::diags() << DgSqlCode(2055)
+      //validate the directory
+      if ((stat(token,&st) != 0 ) &&  !S_ISDIR(st.st_mode) ) //&& (numDirs > MAX_SCRATCH_LOCATIONS))
+        {
+          // syntax error in default, issue a warning (not an error)
+          *CmpCommon::diags() << DgSqlCode(2055)
 			      << DgString0(def)
 			      << DgString1(defName);
-	  // don't continue after a syntax error
-	  str = NULL;
-	}
+          // don't continue after a syntax error
+          str = NULL;
+        }
       else
-	{
-	  tempList.insert(new(heap) ExScratchDiskDrive(
-	       driveLetter,
-	       1, // Thanks to Bill Gates
-	       nodeNum));
-	}
-
-      NADELETEBASIC(driveLetter, heap);
-      driveLetter = NULL;
-
+        {
+          tempList.insert(new(heap) ExScratchDiskDrive(
+                               token,
+                               strlen(token) ));
+        }
+      token = strtok(NULL,sep);
     }
+      
+  NADELETEBASIC(token, heap);
```

Comment:
Changed it to not allocate the string at all. token would just point to the static array. Also changed to use strtok_r. Thank you for pointing this out !